target_chains/ethereum/contracts/contracts/entropy/Entropy.sol

```diff
@@ -253,7 +253,16 @@ abstract contract Entropy is IEntropy, EntropyState {
 
         bytes32 blockHash = bytes32(uint256(0));
         if (req.useBlockhash) {
-            blockHash = blockhash(req.blockNumber);
+            bytes32 _blockHash = blockhash(req.blockNumber);
+
+            // The request can only be stale if the blockhash is zero. The other case when blockhash
+            // can be zero is when the request and reveal is in the same block which can't happen as
+            // the service will reveal only when the request is finalized in a block
+            // Though this will fail for that case too.
```

Comment:
this comment is confusing. I don't know what a "stale request" is, and this comment assumes I do. Then it proceeds to say "well also it fails when the blockhash is 0 for a different reason".

I think the comment should explain why the blockhash can be zero, and why we need this check in those cases.

Also suggest changing the error to be `BlockhashUnavailable`, which is a more precise description of the problem.